external_search.py

```diff
@@ -63,84 +71,112 @@ def bulk(docs, **kwargs):
         self.bulk = bulk
             
         if not self.indices.exists(self.works_index):
-            self.setup_index()
+            self.setup_index(current_alias=use_alias)
 
-    def setup_index(self):
-        """
-        Create the search index with appropriate mapping.
+    def setup_index(self, new_index=None, current_alias=False):
+        """Create the search index with appropriate mapping.
 
         This will destroy the search index, and all works will need
         to be indexed again. In production, don't use this on an
         existing index. Use it to create a new index, then change the 
         alias to point to the new index.
         """
-
-        if self.works_index:
```

Comment:
Despite the diff, this method has barely changed. Only the top three lines and the last three lines are any different. Everything else has just been tabbed left to adjust for the removal of this `if` clause.